### ISSUE TITLE
[Bug]: Installing extensions - unable to get issuer certificate

### ISSUE THREAD

**Issue by ryjogo** [coder/code-server#6051]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS/Web Information

- Remote Architecture: k8s
- `code-server --version`: 4.10.0 

### Steps to Reproduce

1.  start code-server
2. code-server -vvv --auth none --port 13337 --install-extension ms-vscode.powershell 

### Expected

Should install the extensions using the ca's provided..

### Actual

coder@workspaces-rgo-123:~$ export NODE_EXTRA_CA_CERTS=/home/coder/cert.pem
coder@workspaces-rgo-123:~$ code-server -vvv --auth none --port 13337 --install-extension ms-vscode.powershell 
[2023-03-03T09:45:20.909Z] debug Found VS Code arguments; spawning VS Code CLI
[2023-03-03T09:45:20.911Z] debug Running Code CLI
Installing extensions...
unable to get issuer certificate

### Does this issue happen in VS Code or GitHub Codespaces?

- [X] I cannot reproduce this in VS Code.
- [X] I cannot reproduce this in GitHub Codespaces.

### Are you accessing code-server over HTTPS?

- [X] I am using HTTPS.

### Notes

I have tried to use NODE_EXTRA_CA_CERTS but to no avail.

**Comment by code-asher** [comment]
VS Code unfortunately overrides NODE_TLS_REJECT_UNAUTHORIZED.

Any idea if you get similar issues with curl?  Just to rule out 
any Node or code-server issues.  Maybe something like:

```
curl -L https://open-vsx.org/api/ms-vscode/powershell/2023.2.1/file/ms-vscode.powershell-2023.2.1.vsix -o ms-vscode.powershell-2023.2.1.vsix
```

**Comment by ryjogo** [comment]
Yes the curl request works fine, i get the file. I can also install the extension in the VSCode (web)UI.

**Comment by code-asher** [comment]
Oh that is very interesting.  The UI just calls the code-server 
backend for installing extensions so if it works there it should 
work on the command line.  Is there anything different about how 
you spawn the main code-server process compared to your shell?  I 
assume you set NODE_EXTRA_CA_CERTS there as well?

**Comment by ryjogo** [comment]
So spawning the main code-server process is the standard:

```
/usr/bin/code-server --auth none --port 13337 >/tmp/code-server.log 2>&1 &
```

I am not setting `NODE_EXTRA_CA_CERTS` anywhere else actually, i set `HTTP_PROXY`, `HTTPS_PROXY` and `NO_PROXY`.

My intention is however to install the extensions before spawning the main "daemon".

I have all the necessary certs in /etc/ssl/certs for our CA, and we are going through a corporate proxy with ssl interception, hence the need for CA's.

**Comment by ryjogo** [comment]
I just ran with `NODE_DEBUG` and i saw this:

this (masked) proxyserver.local is the proxy server address, so i can see it's connecting through this, however rejectUnauthorized is true?

```
... snip 1 million lines
NET 18505: createConnection [
  {
    host: 'proxyserver.local',
    port: 8080,
    auth: null,
    rejectUnauthorized: true
  },
  null,
  [Symbol(normalizedArgs)]: true
]
NET 18505: pipe false undefined
...
```

**Comment by code-asher** [comment]
Interesting!  I tested `echo '{"http.proxyStrictSSL": false}' > /home/coder/.local/share/code-server/Machine/settings.json` and it appears to not take effect when installing from the command line but it does from the UI.

Putting it in `User/settings.json` does seem to work though.  I am not sure why it seems to ignore `Machine` settings when running from the command line.

Still, even with `rejectUnauthorized` set to true it is strange that it struggles with the certificates.

**Comment by code-asher** [comment]
I am not sure how best to reproduce, I tried `mitmproxy` but I get a different error:

```
coder@dev:~$ rm ~/.local/share/code-server/extensions -r
coder@dev:~$ http_proxy=https://localhost:8000 code-server --install-extension ms-python.python
Installing extensions...
self signed certificate in certificate chain
```

Still, I suppose this does at least show `NODE_EXTRA_CA_CERTS` does get read and used so maybe there is something not quite right with your certificate?  Although if it works for `curl` and friends I am not sure how that could be the case.  :thinking: 

```
coder@dev:~$ http_proxy=https://localhost:8000 NODE_EXTRA_CA_CERTS=$HOME/.mitmproxy/mitmproxy-ca.pem code-server --install-extension ms-python.python
Installing extensions...
Installing extension 'ms-python.python'...
Extension 'ms-python.python' v2023.4.0 was successfully installed.
```

**Comment by code-asher** [comment]
One more thing, VS Code uses `vscode-proxy-agent` which is supposed to read certs from `/etc/ssl/certs/ca-certificates.crt` and `/etc/ssl/certs/ca-bundle.crt` (in addition to whatever is in `NODE_EXTRA_CA_CERTS`):

https://github.com/microsoft/vscode-proxy-agent/blob/49c0f39c327ce408ce69247a3db10a2b313d44a2/src/index.ts#L454-L455

**Comment by ryjogo** [comment]
@code-asher Are you actually using the proxy there though?

i.e. are you sure that code-server is using the http url as you are only setting http_proxy, and not https_proxy too?

coder@dev:~$ http_proxy=https://localhost:8000 NODE_EXTRA_CA_CERTS=$HOME/.mitmproxy/mitmproxy-ca.pem code-server --install-extension ms-python.python
Installing extensions...

**Comment by ryjogo** [comment]
Thanks for the confirmation anyway.. it looks like there is some cert problem this side though, as it works there..

**Comment by code-asher** [comment]
> are you sure that code-server is using the http url

Yeah, I see the requests in mitmproxy.  VS Code falls back to `http_proxy` even for https requests which does seem confusing.

https://github.com/microsoft/vscode-proxy-agent/blob/49c0f39c327ce408ce69247a3db10a2b313d44a2/src/index.ts#L63
